apple/swift-docc-render
# Highlight unique components of overloaded declarations

Bug/issue #, if applicable: 117503463

## Summary

Adds support for highlighting the unique components of overloaded declarations through a new, recursive declaration token component. These highlighted tokens will only be visually rendered when a group of overloaded symbols have been expanded by the user.

#### Example

Here, the highlights emphasize that this function is overloaded by the type of its only parameter.

<img width="1466" alt="Screenshot 2024-05-28 at 5 03 28 PM" src="https://github.com/apple/swift-docc-render/assets/212918/74b0b54c-6755-44e1-a8e0-2d8a7329f5b2">

## Dependencies

`swift-docc`: [Highlight declaration differences in overloaded symbol groups #928](https://github.com/apple/swift-docc/pull/928)

## Testing

Steps:
1. Checkout this branch and run `npm run build`
2. Checkout the branch of `swift-docc` from [this PR](https://github.com/apple/swift-docc/pull/928) and run `swift build`
3. Run `DOCC_HTML_DIR=/path/to/swift-docc-render/dist swift run docc preview --enable-experimental-overloaded-symbol-presentation Tests/SwiftDocCTests/Test\ Bundles/OverloadedSymbols.docc`
4. Verify that the unique parts of overloaded functions like the one in the above example are visually highlighted, only after disclosing the full list of the overloaded variants

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.


\cc @QuietMisdreavus
